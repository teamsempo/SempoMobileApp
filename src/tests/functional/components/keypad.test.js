import React from 'react';
import { shallow } from 'enzyme';
import KeypadComponent from '../../../components/Keypad.js';
import configureMockStore from 'redux-mock-store';

describe('PinComponent', () => {
    describe('Rendering', () => {
        it('should match to snapshot', () => {
            const submitFunction = jest.fn();
            const pinFunction = jest.fn();

            const inputButtons = [
                [1, 2, 3],
                [4, 5, 6],
                [7, 8, 'S'],
            ];
            const component = shallow(
                <KeypadComponent
                    value={''}
                    inputButtons={inputButtons}
                    onChange={(pin) => pinFunction(pin)}
                    onSubmit={() => submitFunction()}
                />
            ).dive();
            expect(component).toMatchSnapshot();
            const button = component.find("InputButton")
            const oneKey = button.at(0);
            const fourKey = button.at(3);
            const eightKey = button.at(7);
            const s_key = button.at(8);
            oneKey.simulate('press');
            fourKey.simulate('press');
            eightKey.simulate('press');
            oneKey.simulate('press');

            expect(submitFunction).not.toHaveBeenCalled();
            s_key.simulate('press');
            expect(submitFunction).toHaveBeenCalled();

            expect(pinFunction.mock.calls[0][0]).toBe('1');
            expect(pinFunction.mock.calls[1][0]).toBe('4');
            expect(pinFunction.mock.calls[2][0]).toBe('8');
            expect(pinFunction.mock.calls[3][0]).toBe('1');
        });
    });
});

