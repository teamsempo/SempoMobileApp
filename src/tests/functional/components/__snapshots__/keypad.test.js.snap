// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PinComponent Rendering should match to snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFF",
      "flex": 8,
    }
  }
>
  <View
    key="row-0"
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <InputButton
      key="0-0"
      onPress={[Function]}
      value={1}
    />
    <InputButton
      key="0-1"
      onPress={[Function]}
      value={2}
    />
    <InputButton
      key="0-2"
      onPress={[Function]}
      value={3}
    />
  </View>
  <View
    key="row-1"
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <InputButton
      key="1-0"
      onPress={[Function]}
      value={4}
    />
    <InputButton
      key="1-1"
      onPress={[Function]}
      value={5}
    />
    <InputButton
      key="1-2"
      onPress={[Function]}
      value={6}
    />
  </View>
  <View
    key="row-2"
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <InputButton
      key="2-0"
      onPress={[Function]}
      value={7}
    />
    <InputButton
      key="2-1"
      onPress={[Function]}
      value={8}
    />
    <InputButton
      key="2-2"
      onPress={[Function]}
      value="S"
    />
  </View>
</View>
`;
